substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/SubstrateOptions.java

```diff
@@ -659,8 +659,17 @@ public static Path getDebugInfoSourceCacheRoot() {
         }
     }
 
-    @Option(help = "Strip debug info from the binary.")//
-    public static final HostedOptionKey<Boolean> StripDebugInfo = new HostedOptionKey<>(true);
+    @Option(help = "Use a separate file for debug info.")//
+    public static final HostedOptionKey<Boolean> StripDebugInfo = new HostedOptionKey<>(OS.getCurrent() != OS.DARWIN, SubstrateOptions::validateStripDebugInfo);
+
+    private static void validateStripDebugInfo(HostedOptionKey<Boolean> optionKey) {
```

Comment:
Ideally this flag should be called `SeparateDebugInfo` rather than `StripDebugInfo` since "strip" means different things on different platform:
* stripping can have an effect on linux even if no debug info was generated (e.g., stripping symbols)
* on windows debug info is always separated to a pdb file. That pdb file can also be ["stripped"](https://learn.microsoft.com/en-us/cpp/build/reference/pdbstripped-strip-private-symbols) to only contain public symbols and stack walking data. Currently that's controlled by `DeleteLocalSymbols` on windows.

I think the PR is already a step in the right direction but maybe later we should look into how to improve/rename those options to have a more uniform meaning on different platforms.